src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/Symmetric/SymmetricOneShotBase.cs

```diff
@@ -0,0 +1,383 @@
+                return;
+            }
+
+            using (SymmetricAlgorithm alg = CreateAlgorithm())
+            {
+                alg.Key = Key;
+
+                Span<byte> destinationBuffer = new byte[plaintext.Length - 1];
+
+                int bytesWritten;
+                bool result = mode switch
+                {
+                    CipherMode.ECB => alg.TryDecryptEcb(ciphertext, destinationBuffer, padding, out bytesWritten),
+                    CipherMode.CBC => alg.TryDecryptCbc(ciphertext, Iv, destinationBuffer, out bytesWritten, padding),
+                    _ => throw new NotImplementedException(),
+                };
+
+                Assert.False(result, "TryDecrypt");
+                Assert.Equal(0, bytesWritten);
+            }
+        }
+
+        protected void TryEncryptOneShot_DestinationTooSmallTest(byte[] plaintext, byte[] ciphertext, PaddingMode padding, CipherMode mode)
+        {
+            if (ciphertext.Length == 0)
+            {
+                // Can't have a too small buffer for zero.
+                return;
+            }
+
+            using (SymmetricAlgorithm alg = CreateAlgorithm())
+            {
+                alg.Key = Key;
+
+                Span<byte> destinationBuffer = new byte[ciphertext.Length - 1];
+
+                int bytesWritten;
+                bool result = mode switch
+                {
+                    CipherMode.ECB => alg.TryEncryptEcb(plaintext, destinationBuffer, padding, out bytesWritten),
+                    CipherMode.CBC => alg.TryEncryptCbc(plaintext, Iv, destinationBuffer, out bytesWritten, padding),
+                    _ => throw new NotImplementedException(),
+                };
+                Assert.False(result, "TryEncrypt");
+                Assert.Equal(0, bytesWritten);
+            }
+        }
+
+        protected void TryDecryptOneShot_DestinationJustRightTest(byte[] plaintext, byte[] ciphertext, PaddingMode padding, CipherMode mode)
+        {
+            using (SymmetricAlgorithm alg = CreateAlgorithm())
+            {
+                alg.Key = Key;
+
+                int expectedPlaintextSize = padding == PaddingMode.Zeros ? ciphertext.Length : plaintext.Length;
+                Span<byte> destinationBuffer = new byte[expectedPlaintextSize];
+
+                int bytesWritten;
+                bool result = mode switch
+                    {
+                        CipherMode.ECB => alg.TryDecryptEcb(ciphertext, destinationBuffer, padding, out bytesWritten),
+                        CipherMode.CBC => alg.TryDecryptCbc(ciphertext, Iv, destinationBuffer, out bytesWritten, padding),
+                        _ => throw new NotImplementedException(),
+                    };
+                Assert.True(result, "TryDecrypt");
+                Assert.Equal(destinationBuffer.Length, bytesWritten);
+
+                AssertPlaintexts(plaintext, destinationBuffer.ToArray(), padding);
```

Comment:
Seems like the helpers could be spanified.  But doesn't really matter.

Reply:
Or just split the array/span versions of destination:

```diff
-Span<byte> destinationBuffer = new byte[expectedPlaintextSize];
+byte[] destination = new byte[expectedPlaintextSize];
+Span<byte> destinationBuffer = destination;
```

Reply:
I kept the assert helpers as taking arrays because asserting on arrays is nicer with Xunit. Taking a `ReadOnlySpan<byte>` would mean doing an `Assert.True(blah.SequenceEqual(otherBlah))` which doesn't have quite as useful output when the assertion fails.

Reply:
Yeah.  And the version for strings is nicer than the one for arrays (particularly `byte[]`)... at least in the past the array mismatch one was most useful if it was in the first few elements.  That's why there are so many Asserts over foo.ByteArrayToHex().

One of these days I should make an assert-equality routine and clean up all of the things to do non-allocated SequenceEqual then fall back to making the string on failure.  Maybe it'd shave a non-trivial amount of time off of the test execution.

Reply:
Oh. Actually it looks like we have one on `AssertExtensions`. Let me play with that a bit...